### ISSUE TITLE
[Question] The loss first normal, but suddenly become 0

### ISSUE THREAD

**Issue by lucasjinreal** [haotian-liu/LLaVA#1231]
### Question

```
{'loss': 1.0176, 'learning_rate': 1.6180582605257102e-05, 'epoch': 0.62}                                                                                                                                                        
{'loss': 0.9101, 'learning_rate': 1.6180267518266783e-05, 'epoch': 0.62}                                                                                                                                                        
{'loss': 0.8977, 'learning_rate': 1.6179952421348435e-05, 'epoch': 0.62}                                                                                                                                                        
{'loss': 0.948, 'learning_rate': 1.6179637314502555e-05, 'epoch': 0.62}                                                                                                                                                         
{'loss': 1.2777, 'learning_rate': 1.6179322197729653e-05, 'epoch': 0.62}                                                                                                                                                        
{'loss': 1.3493, 'learning_rate': 1.6179007071030236e-05, 'epoch': 0.62}                                                                                                                                                        
{'loss': 1.3082, 'learning_rate': 1.6178691934404813e-05, 'epoch': 0.62}                                                                                                                                                        
{'loss': 1.0649, 'learning_rate': 1.6178376787853884e-05, 'epoch': 0.62}                                                                                                                                                        
{'loss': 1.4151, 'learning_rate': 1.6178061631377956e-05, 'epoch': 0.62}                                                                                                                                                        
{'loss': 1.0117, 'learning_rate': 1.6177746464977546e-05, 'epoch': 0.62}                                                                                                                                                        
{'loss': 0.975, 'learning_rate': 1.617743128865314e-05, 'epoch': 0.62}                                                                                                                                                          
{'loss': 0.9362, 'learning_rate': 1.6177116102405265e-05, 'epoch': 0.62}                                                                                                                                                        
{'loss': 1.0906, 'learning_rate': 1.6176800906234415e-05, 'epoch': 0.62}                                                                                                                                                        
{'loss': 0.873, 'learning_rate': 1.61764857001411e-05, 'epoch': 0.62}                                                                                                                                                           
{'loss': 1.0242, 'learning_rate': 1.617617048412583e-05, 'epoch': 0.62}                                                                                                                                                         
{'loss': 0.8448, 'learning_rate': 1.6175855258189102e-05, 'epoch': 0.62}                                                                                                                                                        
{'loss': 1.0232, 'learning_rate': 1.6175540022331428e-05, 'epoch': 0.62}                                                                                                                                                        
{'loss': 0.7128, 'learning_rate': 1.6175224776553317e-05, 'epoch': 0.62}                                                                                                                                                        
{'loss': 0.9428, 'learning_rate': 1.6174909520855272e-05, 'epoch': 0.62}                                                                                                                                                        
{'loss': 0.9259, 'learning_rate': 1.61745942552378e-05, 'epoch': 0.62}                                                                                                                                                          
{'loss': 0.8517, 'learning_rate': 1.617427897970141e-05, 'epoch': 0.62}                                                                                                                                                         
{'loss': 0.0, 'learning_rate': 1.61739636942466e-05, 'epoch': 0.62}                                                                                                                                                             
{'loss': 0.0, 'learning_rate': 1.6173648398873888e-05, 'epoch': 0.62}                                                                                                                                                           
{'loss': 0.0, 'learning_rate': 1.617333309358377e-05, 'epoch': 0.62}                                                                                                                                                            
{'loss': 0.0, 'learning_rate': 1.6173017778376762e-05, 'epoch': 0.62}                                                                                                                                                           
{'loss': 0.0, 'learning_rate': 1.6172702453253365e-05, 'epoch': 0.62} 
```

the loss first normal, but suddenly got 0, training on stage2. Base model is qwen2

it was actually worked on model at 0.5 epoch, but got 0 later.

Why

**Comment by patrick-tssn** [comment]
when I change zero3 to zero2-offload, I meet the same problem, and my base model is Vicuna.

**Comment by patrick-tssn** [comment]
In my case, setting the 'overlap_comm' parameter to false resulted in the disappearance of the phenomenon. This issue may originate from DeepSpeed, although I cannot confirm this with certainty. Inspired by [this issue](https://github.com/microsoft/DeepSpeed/issues/4298)

**Comment by lucasjinreal** [comment]
@patrick-tssn Hi, you mean, change:

```
"zero_optimization": {
        "stage": 2,
        "overlap_comm": true,
        "contiguous_gradients": true,
        "sub_group_size": 1e9,
        "reduce_bucket_size": "auto"
    }
```

this to false? 

Have u tested if the training can result normal loss and normal result?

**Comment by patrick-tssn** [comment]
> @patrick-tssn Hi, you mean, change:
> 
> ```
> "zero_optimization": {
>         "stage": 2,
>         "overlap_comm": true,
>         "contiguous_gradients": true,
>         "sub_group_size": 1e9,
>         "reduce_bucket_size": "auto"
>     }
> ```
> 
> this to false?
> 
> Have u tested if the training can result normal loss and normal result?

Yes, setting `"overlap_comm": false` worked for me; I am now getting the normal loss. However, I haven't evaluated it yet as I am still in the process of training.

**Comment by lucasjinreal** [comment]
I am tested same zero2 config on finetuning whole model directly with SAM & a Opt model except `overlap_comm`, the loss with `overlap_comm` set to True always exceed to maximum value (Nan) while false loss seems can decrease now.

And the first iter loss doubled if set to True.

Any reason for the differences? Looks like for any full-model task, if using zero2, it should overlap_coomn to False
